core/src/main/java/com/datastax/oss/driver/api/core/cql/PagingState.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.core.cql;
+
+import com.datastax.oss.driver.api.core.session.Session;
+import com.datastax.oss.driver.internal.core.cql.DefaultPagingState;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.nio.ByteBuffer;
+
+/**
+ * A safe wrapper around the paging state of a query.
+ *
+ * <p>This class performs additional checks to fail fast if the paging state is not reused on the
+ * same query, and it provides utility methods for conversion to/from strings and byte arrays.
+ */
+public interface PagingState {
+
+  /** Parses an instance from a string previously generated with {@code toString()}. */
+  @NonNull
+  static PagingState fromString(@NonNull String string) {
+    return DefaultPagingState.fromString(string);
```

Comment:
It seems a bit odd that the interface is referring to the method that is on the implementation. I wonder if that dependency may introduce some problems in the future?

Reply:
I don't think so, the public contract only states that it will return an instance of the interface, the actual class is an implementation detail.
We already do this in a number of places (`SimpleStatement.newInstance`, etc).